### PR TITLE
STYLE-6 forms component added, and a few other minor tweaks elsewhere

### PR DESCRIPTION
**Pull request by hrag**

### CODE REVIEW

File: src/docs/components/navigation.jsx [hunk 1]
```diff
@@ -3,26 +3,28 @@ import AnchorLink from 'react-anchor-link-smooth-scroll'
...
-        this.sectionClickHandler = this.sectionClickHandler.bind(this);
+        super(props)
+        this.sectionClickHandler = this.sectionClickHandler.bind(this)
     }
 
-    sectionClickHandler = (e) => {
-        const current = e.target.attributes['section-id'].value;
-        const that = this;
+    sectionClickHandler = e => {
+        const current = e.target.attributes['section-id'].value
```
Comment (chalisegrogan) [inline src/docs/components/navigation.jsx]: Why did semicolons get removed?

File: src/docs/components/navigation.jsx [hunk 2]
```diff
@@ -46,62 +52,64 @@ class Nav extends React.Component {
...
-                            className="nav-link" 
-                            role="link" 
+                        <AnchorLink
+                            href="#getting-started-section"
+                            className="nav-link"
+                            role="link"
                             offset="-42"
-                            aria-label="Getting Started" 
-                            onClick={this.sectionClickHandler} 
+                            aria-label="Getting Started"
```
Comment (chalisegrogan) [inline src/docs/components/navigation.jsx]: Yay all these are removing excess whitespace 😍

File: src/docs/sections/buttons.jsx [hunk 3]
```diff
@@ -1,63 +1,84 @@
...
+                </div>
+                <p>
+                    <strong>Options</strong>
+                </p>
                 <ul>
-                    <li><strong>Color:</strong> <em>blue (default), green, pink, red, gray, white, disabled. You can add additional colors in the mixin if needed.</em></li>
-                    <li><strong>Size:</strong> <em>large (default), medium, small</em></li>
-                    <li><strong>Type:</strong> <em>solid (default), outlined</em></li>
+                    <li>
+                        <strong>Color:</strong>{' '}
```
Comment (chalisegrogan) [inline src/docs/sections/buttons.jsx]: Ugh, JSX not converting newlines to whitespace for inline elements (like HTML does) really irritates me.

File: src/docs/sections/buttons.jsx [hunk 4]
```diff
@@ -66,64 +87,122 @@ class Buttons extends React.Component {
...
                 <p>Large and medium sizes</p>
                 <div className="has-code">
-                    For large buttons, height is 48px, font size is <InlineCode>$sd-base-text</InlineCode> (16px), left and right padding is 24px with a minimum width of 120px.
+                    For large buttons, height is 48px, font size is{' '}
+                    <InlineCode>$sd-base-text</InlineCode> (16px), left and
+                    right padding is 24px with a minimum width of 120px.
                 </div>
                 <div className="has-code">
-                    For medium buttons, height is 36px, font size is <InlineCode>$sd-x-small-text</InlineCode> (14px), left and right padding is 18px with a minimum width of 72px.
+                    For medium buttons, height is 36px, font size is{' '}
```
Comment (chalisegrogan) [inline src/docs/sections/buttons.jsx]: so I think you can accomplish _some_ of this without the `{' '}` by avoiding putting inline elements at the start or end of the line. For example, this should work:

```jsx
<div>
    For medium buttons, height is 36px, font size
    is <InlineCode>$sd-x-small-text</InlineCode> (14px), 
    left and right padding is 18px with a minimum width of 72px.
</div>
```

File: src/docs/sections/colors.jsx [hunk 5]
```diff
@@ -9,15 +9,23 @@ class Colors extends React.Component {
                 className="sg-section sg-colors"
                 id="colors-section"
                 ref={this.props.innerRef}
-                tabIndex={`${this.props.currentSection === 'colors' ? '-1' : ''}`}
+                tabIndex={`${
```
Comment (chalisegrogan) [inline src/docs/sections/colors.jsx]: This also doesn't need string interpolation.

File: src/docs/sections/colors.jsx [hunk 6]
```diff
@@ -52,13 +60,15 @@ class Colors extends React.Component {
                             $periwinkle <span className="def">#3595D6</span>
                         </div>
                         <div className="dark-periwinkle">
-                            $dark-periwinkle <span className="def">#005B98</span>
+                            $dark-periwinkle{' '}
```
Comment (chalisegrogan) [inline src/docs/sections/colors.jsx]: Curious why you chose to have this on two lines (which requires the `{' '}`) but not consistently (like `$ultra-marine` below). I'd vote for keeping it all on one line if possible.

Comment (hrag) [reply src/docs/sections/colors.jsx]: Um, I have no idea how this happened... I want to blame some Prettier glitch? I hadn't even opened this file for this branch. Odd... Either way, fixed it. Nice catch.

File: src/docs/sections/colors.jsx (hunk 5 above)
Comment (hrag) [reply src/docs/sections/colors.jsx]: Sweet, thanks! Updated for all sections.

File: src/docs/components/navigation.jsx (hunk 1 above)
Comment (hrag) [reply src/docs/components/navigation.jsx]: Good question. This was automated, probably by Prettier.

File: src/docs/sections/buttons.jsx (hunk 4 above)
Comment (hrag) [reply src/docs/sections/buttons.jsx]: Went through and made some changes like this. In some places, I had to use a space so I went with &nbsp;

File: src/docs/sections/buttons.jsx [hunk 7]
```diff
@@ -66,64 +85,120 @@ class Buttons extends React.Component {
...
+                        <div>
+                            Blue buttons are primary actions within the product.
+                            These actions drive the experience forward.
+                        </div>
                     </div>
 
                     <div className="sg-buttons-sidebyside example-box">
-                        <div className="left-side"><button className="pink-primary">Pink Primary</button></div>
-                        <div>Pink buttons are used to indicate a “sale” action like “add to cart” or “upgrade.”</div>
+                        <div className="left-side">
```
Comment (kirstenburns) [inline src/docs/sections/buttons.jsx]: Wondering if you could add somewhere here that we have an already created class for this: 'sd-cta-button' (it makes the pink, large sized button)

Comment (kirstenburns) [reply src/docs/sections/buttons.jsx]: or add below where the "sd-button" class is mentioned?

File: src/docs/sections/buttons.jsx [hunk 8]
```diff
@@ -133,110 +208,177 @@ class Buttons extends React.Component {
                 <h2 className="subsection-title">Button styles</h2>
                 <h3>Reusable classes</h3>
                 <p>
-                    The blue primary button is the most used button style and we should not recreate it with a mixin every time it is needed.
+                    The blue primary button is the most used button style and we
+                    should not recreate it with a mixin every time it is needed.
                 </p>
                 <div className="has-code">
-                    Use <InlineCode>sd-button</InlineCode> class to create the blue primary button.
+                    Use <InlineCode>sd-button</InlineCode> class to create the
```
Comment (kirstenburns) [inline src/docs/sections/buttons.jsx]: could also add reusable class: 'sd-secondary-button`

**Pull request merged**